dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs

```diff
@@ -0,0 +1,162 @@
+    /// Integration test for <see cref="OpenAIAssistantAgent"/> using function calling
+    /// and targeting Open AI services.
+    /// </summary>
+    [Fact(Skip = "OpenAI will often throttle requests. This test is for manual verification.")]
+    public async Task OpenAIMixedAgentTestAsync()
+    {
+        OpenAIConfiguration openAISettings = this._configuration.GetSection("OpenAI").Get<OpenAIConfiguration>()!;
+        Assert.NotNull(openAISettings);
+
+        await this.ExecuteAgentAsync(
+            this.CreateChatCompletionKernel(openAISettings),
+            new(openAISettings.ApiKey),
+            AssistantModel);
+    }
+
+    /// <summary>
+    /// Integration test for <see cref="OpenAIAssistantAgent"/> using function calling
+    /// and targeting Azure OpenAI services.
+    /// </summary>
+    [Fact]
+    public async Task AzureOpenAIMixedAgentAsync()
+    {
+        AzureOpenAIConfiguration azureOpenAISettings = this._configuration.GetSection("AzureOpenAI").Get<AzureOpenAIConfiguration>()!;
+        Assert.NotNull(azureOpenAISettings);
+
+        await this.ExecuteAgentAsync(
+            this.CreateChatCompletionKernel(azureOpenAISettings),
+            new(azureOpenAISettings.ApiKey, azureOpenAISettings.Endpoint),
+            azureOpenAISettings.ChatDeploymentName!);
+    }
+
+    private async Task ExecuteAgentAsync(
+        Kernel chatCompletionKernel,
+        OpenAIAssistantConfiguration config,
+        string modelName)
+    {
+        // Arrange
+        KernelPlugin plugin = KernelPluginFactory.CreateFromType<MenuPlugin>();
+
+        // Configure chat agent with the plugin.
+        ChatCompletionAgent chatAgent =
+            new()
+            {
+                Kernel = chatCompletionKernel,
+                Instructions = "Answer questions about the menu.",
+                Arguments = new(new OpenAIPromptExecutionSettings() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions }),
+            };
+        chatAgent.Kernel.Plugins.Add(plugin);
+
+        // Assistant doesn't need plug-in since it has access to the shared function result.
+        OpenAIAssistantAgent assistantAgent =
+            await OpenAIAssistantAgent.CreateAsync(
+                kernel: new(),
+                config,
+                new()
+                {
+                    Instructions = "Answer questions about the menu.",
+                    ModelId = modelName,
+                });
+
+        // Act
+        try
+        {
+            AgentGroupChat chat = new(chatAgent, assistantAgent);
+            await this.InvokeAgentAsync(chat, chatAgent, "What is the special soup?", "Clam Chowder");
+            await this.InvokeAgentAsync(chat, assistantAgent, "What is the special drink?", "Chai Tea");
+        }
+        finally
+        {
+            await assistantAgent.DeleteAsync();
+        }
+    }
+
+    private async Task InvokeAgentAsync(AgentGroupChat chat, Agent agent, string input, string expected)
```

Comment:
Is not clear that Asserting is happening within this function, suggest adding `Assert` as a prefix.

```suggestion
    private async Task AssertInvokeAgentAsync(AgentGroupChat chat, Agent agent, string input, string expected)
```